cmd/admin-heal-ops.go

```diff
@@ -625,14 +626,35 @@ func (h *healSequence) healBucket(bucket string) error {
 	marker := ""
 	isTruncated := true
 	for isTruncated {
+		if globalHTTPServer != nil {
+			// Wait at max 1 minute for an inprogress request
+			// before proceeding to heal
+			waitCount := 60
+			// Any requests in progress, delay the heal.
+			for globalHTTPServer.GetRequestCount() > 0 && waitCount > 0 {
+				waitCount--
+				time.Sleep(1 * time.Second)
+			}
+		}
+
+		// Heal 256 objects at a time.
 		objectInfos, err := objectAPI.ListObjectsHeal(h.ctx, bucket,
-			h.objPrefix, marker, "", 1000)
+			h.objPrefix, marker, "", 256)
```

Comment:
Any reason for 256 here? It looks arbitrary and seems high for small number of node deployments.

At least 1 cpu is needed to actively heal an object and trying to heal many simultaneously is not going to be faster, as most will simply wait for cpu/disk. It seems like `numCPU * numNodes` is a better degree of parallelism. WIth a large number of healing threads, normal processing will be slowed down (after the 1 minute delay is done).

Reply:
256 is simply an arbitrary number. Perhaps we can do that runtime.CPU() * len(globalEndpoints)